Conflict when using quarkus-kafka-messaging and quarkus-spring-di

When I am using quarkus-kafka-messaging and quarkus-spring-di dependencies I am getting the following error:


```shell
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.arc.deployment.ArcProcessor#registerBeans threw an exception: jakarta.enterprise.inject.spi.DefinitionException: Bean class dev.matheuscruz.quarkusclub.product.ProductConsumer declares multiple scope type annotations: jakarta.enterprise.context.Dependent, jakarta.inject.Singleton
```

It happens with the following code:

```java
package dev.matheuscruz.quarkusclub.product;

import org.eclipse.microprofile.reactive.messaging.Incoming;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Service;

@Service
// we have conflict between @Dependent generated by **quarkus-kafka-messaging** and @Singleton generated by **quarkus-spring-di**
public class ProductConsumer {

    private static final Logger LOGGER = LoggerFactory.getLogger(ProductConsumer.class);

    @Incoming("product.created.v1")
    public void consume(String message) {
        LOGGER.info("📥 Received message from Kafka about new product: {}", message);
    }
}
```

I think we have conflict between `@Dependent` generated by **quarkus-kafka-messaging** and `@Singleton` generated by **quarkus-spring-di**.

--- a/extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/GrpcServerProcessor.java
+++ b/extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/GrpcServerProcessor.java
@@ -492,9 +492,16 @@ public class GrpcServerProcessor {
                     }
 
                     @Override
+                    public int priority() {
+                        // Must run after the SpringDIProcessor annotation transformer, which has default priority 1000
+                        return 500;
+                    }
+
+                    @Override
                     public void transform(TransformationContext context) {
                         ClassInfo clazz = context.getTarget().asClass();
-                        if (userDefinedServices.contains(clazz.name()) && !customScopes.isScopeDeclaredOn(clazz)) {
+                        if (userDefinedServices.contains(clazz.name())
+                                && !customScopes.isScopeIn(context.getAnnotations())) {
                             // Add @Singleton to make it a bean
                             context.transform()
                                     .add(BuiltinScope.SINGLETON.getName())

--- a/extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/SmallRyeReactiveMessagingProcessor.java
+++ b/extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/SmallRyeReactiveMessagingProcessor.java
@@ -135,11 +135,17 @@ public class SmallRyeReactiveMessagingProcessor {
             }
 
             @Override
+            public int priority() {
+                // Must run after the SpringDIProcessor annotation transformer, which has default priority 1000
+                return 500;
+            }
+
+            @Override
             public void transform(AnnotationsTransformer.TransformationContext ctx) {
                 if (ctx.isClass()) {
                     ClassInfo clazz = ctx.getTarget().asClass();
                     Map<DotName, List<AnnotationInstance>> annotations = clazz.annotationsMap();
-                    if (scopes.isScopeDeclaredOn(clazz)
+                    if (scopes.isScopeIn(ctx.getAnnotations())
                             || annotations.containsKey(ReactiveMessagingDotNames.JAXRS_PATH)
                             || annotations.containsKey(ReactiveMessagingDotNames.REST_CONTROLLER)
                             || annotations.containsKey(ReactiveMessagingDotNames.JAXRS_PROVIDER)) {

